data/posts/2017-06-23-expanding-on-phalcon-3-2-0-new-features.md

```diff
@@ -0,0 +1,328 @@
+## Expanding on Phalcon 3.2.0 new features
+
+With [Phalcon 3.2.0](https://blog.phalconphp.com/post/phalcon-3-2-0-released-and-new-docs) there were many new features and bugs fixed. Today we will write about most important things you need to know and show some code examples of new features.
+
+- ##### Added Factory Adapter loaders [#11001](https://github.com/phalcon/cphalcon/issues/11001)
+  With this feature you can load your services in more simplified way using for example ini file:
+  
+  ```ini
+  [database]
+  host = TEST_DB_MYSQL_HOST
+  username = TEST_DB_MYSQL_USER
+  password = TEST_DB_MYSQL_PASSWD
+  dbname = TEST_DB_MYSQL_NAME
+  port = TEST_DB_MYSQL_PORT
+  charset = TEST_DB_MYSQL_CHARSET
+  adapter = mysql
+  ```
+  
+  ```php
+  <?php
+  
+  use Phalcon\Config\Adapter\Ini;
+  use Phalcon\Di;
+  use Phalcon\Db\Adapter\Pdo\Factory;
+
+  $di = new Di();
+  $config = new Ini('config.ini');
+  
+  $di->set('config', $config);
+  
+  $di->set('db', function() {
+      return Factory::load($this->config->database);
+  });
+  ```
+  
+  This properly create your database instance and you can change database adapter to other in ini file anytime without changing your code.
```

Comment:
This will create your database instance and you can change the database adapter to another in the ini file anytime is needed without changing your code.